### ISSUE TITLE
[android] Silence other sounds/music when announcing TTS directions

### ISSUE THREAD

**Issue by biodranik** [organicmaps/organicmaps#5109]
Should it be configurable or not? By default, if user enabled voice directions, he/she expects to hear them. So it is likely should be a default behavior.

**Comment by kosh2323** [comment]
In general, it is better to tune in, turn off all extraneous sounds, or mute the volume, or pause, for example, an audio player.

**Comment by lapaz17** [comment]
System uses AUDIOFOCUS_GAIN_TRANSIENT_MAY_DUCK right now, are we expecting [AUDIOFOCUS_GAIN_TRANSIENT_EXCLUSIVE](https://developer.android.com/reference/android/media/AudioManager.html#AUDIOFOCUS_GAIN_TRANSIENT_EXCLUSIVE) as well? Users might choose among 2 of them?

**Comment by kosh2323** [comment]
It seems that the changes are not worth waiting for....

**Comment by kosh2323** [comment]
I've been waiting for this audi focus option to be added, the program has already been updated, if I'm not mistaken, 3 times, but the developer stubbornly ignores the wishes of users. It is a pity that the project has no future. I delete the program as unnecessary and for hopelessness.

**Comment by biodranik** [comment]
@kosh2323 we, as developers, have a very hard choice of which issue out of more than 1700 should be fixed first. Considering that we're working in our free time, on a free, donation-based project.

@kosh2323 the sound should already duck, e.g. the other music/sound volume should be lowered when OM TTS directions are announced. Is it your case? Is the other sound volume lowered?

**Comment by kosh2323** [comment]
Thank you for the answer, I have no complaints against you, and I understand that developers always have one drawback - it's not enough time, especially for free projects. With respect. Success.

**Comment by biodranik** [comment]
@kosh2323 can you please clarify, does ducking (lowering other sounds volume) work in your case with Organic Maps?

**Comment by kosh2323** [comment]
To be honest, I didn't try, I just waited with updates that such a setting would appear, but I never saw it. Therefore, I decided that the development of the program is going its own way. For the user, everything should be extremely clear.

**Comment by biodranik** [comment]
Wait, you asked for a feature without even trying if it already works for you?

**Comment by kosh2323** [comment]
Tried it a few times before the last update, and thought you were asking if it worked after the last update. I tried it until the last update and it doesn't work, and how should I understand that there is such an option if it is not in the settings. I use several navigation programs, and I see such options in the settings of these programs. But this is not in yours, and I thought that nothing has changed since the last update.

**Comment by biodranik** [comment]
@kosh2323 sorry, it is still not clear to me. What is the current behavior of Organic Maps voice directions on your device, when some other music is playing? Is music volume lowered when Organic Maps announces directions? Also, are you testing it using Bluetooth or without Bluetooth?

**Comment by kosh2323** [comment]
Tested on an 8-inch Samsung TabA tablet. Music that plays simultaneously with the running OrganicMaps program does not reduce the volume and does not pause, as for example another CityGuide navigator does. During his voice prompts, he intercepts the audio focus, and pauses the music, or mutes the volume of his choice in the settings.

**Comment by biodranik** [comment]
Thanks for testing it! You didn't use Bluetooth, right?
It means that the current ducking implementation doesn't work properly. 

Can anyone else confirm that on other devices? @lapaz17

**Comment by kosh2323** [comment]
> Thanks for testing it! You didn't use Bluetooth, right? It means that the current ducking implementation doesn't work properly.
> 
> Can anyone else confirm that on other devices? @lapaz17

Hi, no bluetooth was not used.

**Comment by Switch123456789** [comment]
Hey, I noticed today that it isn't working with Android Auto. However it works fine if I use the phone as is. Let me know if I can help with some testing.

**Comment by biodranik** [comment]
@Switch123456789 what exactly is not working? Ducking or no TTS at all?

**Comment by biodranik** [comment]
@Switch123456789 was your device connected using a cable or using wifi/Bluetooth?

**Comment by Switch123456789** [comment]
> was your device connected using a cable or using wifi/Bluetooth?
It was connected via cable but it always establishes a BT-connection automatically if I plug the cable. (I guess that's normal?)

**Comment by biodranik** [comment]
Let's try to debug this issue. Am I correctly understanding that the voice coming from the phone (not connected to the car audio via cable or Bluetooth) works fine?

1. What about the phone connected via Bluetooth (without Android Auto)? Does it work? Are announcements ducking other sounds?
2. Is the issue the same with Android Auto?

It may come from Bluetooth audio connection, unrelated to Android Auto.

**Comment by Switch123456789** [comment]
I did not have an opportunity to do tests yet but I will within the next few days and check it all out.

**Comment by Switch123456789** [comment]
I was able to do some testing today, with a normal bluetooth connection it works perfectly fine. However, with Android Auto it does not, the music keeps blasting while the voice gives directions.

**Comment by biodranik** [comment]
@Switch123456789 
1. Android phone connected via Bluetooth to car's audio works fine.
2. Android phone connected via Bluetooth to car's audio and via cable to Android Auto doesn't work

Correct? What if Bluetooth is disabled on the phone? Does TTS work without it, via cable only?

**Comment by Switch123456789** [comment]
Correct! It won't let me turn off Bluetooth as long as the cable is connected. If I disconnect the cable, delete the car from known devices, turn Bluetooth off and THEN reconnect the cable, BT turns back on and pairs with the car on it's own. As far as I know, Android Auto needs both, USB & Bluetooth combined.

**Comment by pastk** [comment]
Related:
- previous/current (imperfect) fix https://github.com/organicmaps/organicmaps/pull/901
- https://github.com/organicmaps/organicmaps/issues/4726

**Comment by biodranik** [comment]
Issue when music is not silenced at the time of navigation directions announcement should already be fixed in the latest version. 

@Switch123456789 @kosh2323 would you please confirm that it works for you now? Or report any other issues in your configuration?

**Comment by Switch123456789** [comment]
Unfurtunately I can't test it atm. If I come around to do so I will report back but it might take forever.
